plugin/pkg/shoot/quotavalidator/admission.go

```diff
@@ -238,20 +240,9 @@ func (q *QuotaValidator) Validate(ctx context.Context, a admission.Attributes, o
 			return apierrors.NewInternalError(err)
 		}
 
-		oldLifetime, exists := oldShoot.Annotations[common.ShootExpirationTimestamp]
-		if !exists {
-			// The old version of the Shoot has no clusterLifetime annotation yet.
-			// Therefore we have to calculate the lifetime based on the maxShootLifetime.
-			oldLifetime = oldShoot.CreationTimestamp.Time.Add(time.Duration(*maxShootLifetime*24) * time.Hour).Format(time.RFC3339)
-		}
-		oldExpirationTime, err = time.Parse(time.RFC3339, oldLifetime)
-		if err != nil {
-			return apierrors.NewInternalError(err)
-		}
-
-		maxPossibleExpirationTime = oldExpirationTime.Add(time.Duration(*maxShootLifetime*24) * time.Hour)
+		maxPossibleExpirationTime = q.time.Now().Add(time.Duration(*maxShootLifetime*24) * time.Hour)
```

Comment:
I am not sure, but isn't it possible with this logic to come back for example after one day and prolong the Shoot's lifetime by another day?
Shouldn't we use `shoot.CreationTimestamp` instead of `time.Now()`?

Reply:
AFAIK this is desired behaviour. We don't want to forbid extending the lifetime. We wanted to always allow it based on the configured `Quota`'s `clusterLifeTimeDays` field. The only thing we wanted to ensure is that it must be user-initiated. Hence, if an end-user requires his trial cluster to live longer he can extend it.
Yes, you can come back after a day, but then you have to do it explicitly. We never had a setting that says something like "extending the life time is only allowed three days before it expires". Consequently, I think what @ialidzhikov implemented is correct.

Reply:
cc @dkistner as initial contributor of this feature

Reply:
Alright, then i guess it's perfectly fine :)

Reply:
@ialidzhikov any feedback on this?

Reply:
> Shouldn't we use shoot.CreationTimestamp instead of time.Now()?

I think the desired and described behaviour in #81 is to allow lifetime to be extended (on user demand and action) only up to `.spec. clusterLifeTimeDays` from the current date. If we use  `metadata.creationTimestamp`, how we would allow lifetime to be extended?

The issue I am trying to fix with this PR is to allow only lifetime updates up to `time.Now() + .spec.clusterLifetimeDays`, but never more. See #1837 for a hack how you can currently extend your shoot lifetime to infinity.

/cc @dkistner for feedback

Reply:
Thanks for clarifying! 👍

Reply:
Yes the intention was to allow extending the cluster lifetime from now + max the amount of days configured in `.spec.clusterLifeTimeDays`. Extending to infinity was not intended. Nice catch and thanks for fixing it.